### PR TITLE
Unit tests for `location_table` access pattern

### PR DESCRIPTION
**Pull request by fandreuz**
As discussed under #27, I'm adding some unit tests for the updated access pattern to `location_table`.

I introduced a new function `collect_frame_list` whose job is to access the proper indexes in `location_table` and build a `Vec<Frame>`. I made it a generic to simplify writing unit tests.

### CODE REVIEW

File: src/collector/otlp/service.rs [hunk 1]
```diff
@@ -376,3 +361,80 @@ fn process_sample(
...
+    use itertools::Itertools;
+
+    use super::*;
+
+    #[test]
+    fn sample_frame_list() -> Result<(), Status> {
+        let loc_mapping = (0..11).collect_vec();
+        let location_indices = vec![4, 9, 6, 2, 7, 4, 4, 2, 0, 1, 2, 3, 5];
+
+        assert_eq!(
```
Comment (rockdaboot) [inline src/collector/otlp/service.rs:406]: Add comments/names to the test about what they are testing, e.g.
"2 items from start", "0 items" or "all items"

Add two more tests: "2 items from end", "2 item from middle"

Comment (rockdaboot) [reply src/collector/otlp/service.rs:406]: Not a rust guy here... maybe adding the "names" as error message to the assert_eq macros is better!?

File: src/collector/otlp/service.rs [hunk 2]
```diff
@@ -376,3 +361,80 @@ fn process_sample(
...
+        );
+
+        Ok(())
+    }
+
+    #[test]
+    fn sample_frame_list_err() -> Result<(), Status> {
+        let loc_mapping = (0..11).collect_vec();
+
+        let loc_indices_oob = collect_frame_list(0, 3, &loc_mapping, &vec![0i32, 1i32]);
```
Comment (rockdaboot) [inline src/collector/otlp/service.rs:426]: Add more tests for corner cases, e.g.
```
let loc_indices_oob = collect_frame_list(1, 2, &loc_mapping, &vec![0i32, 1i32]);
let loc_indices_oob = collect_frame_list(2, 0, &loc_mapping, &vec![0i32, 1i32]);
```

File: src/collector/otlp/service.rs [hunk 3]
```diff
@@ -376,3 +361,80 @@ fn process_sample(
 
     Ok(())
 }
+
+fn collect_frame_list<V>(
+    loc_start: usize,
+    loc_len: usize,
+    loc_mapping: &Vec<V>,
```
[the remark this answers is not in the document]
Comment (fandreuz) [reply src/collector/otlp/service.rs:368]: Yeah I agree on this, if there is no strong preference from anybody to keep this naming I'd be happy to rename it to `loc_table`

File: src/collector/otlp/service.rs (hunk 2 above)
Comment (fandreuz) [reply src/collector/otlp/service.rs]: Sure: 14198b81ef3f4b01da5a139ef16ed7b638bc3bfb

File: src/collector/otlp/service.rs (hunk 1 above)
Comment (fandreuz) [reply src/collector/otlp/service.rs:406]: Something like this? 14198b81ef3f4b01da5a139ef16ed7b638bc3bfb

Comment (rockdaboot) [reply src/collector/otlp/service.rs:406]: :+1:

File: src/collector/otlp/service.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (rockdaboot) [reply src/collector/otlp/service.rs:368]: We can do this in a separate PR.

**Comment by fandreuz** [comment]
@rockdaboot is this good to go?
